server/tables/tables.go

```diff
@@ -620,20 +631,28 @@ func postMigrateInvocationUUIDForMySQL(db *gorm.DB) error {
 			(SELECT invocation_uuid FROM Invocations i 
 			WHERE i.invocation_pk=ts.invocation_pk)
 		WHERE invocation_uuid IS NULL`
-	return updateInBatches(db, updateTargetStatusStmt, 10000)
+	if err := updateInBatches(db, updateTargetStatusStmt, 10000); err != nil {
+		return err
+	}
+	return db.Exec("ALTER TABLE TargetStatuses DROP PRIMARY KEY, ADD PRIMARY KEY(target_id, invocation_uuid) ").Error
```

Comment:
According to the mysql docs [1], you should add `ALGORITHM=INPLACE, LOCK=NONE` to the alter statement to make sure it does not lock. The documentation doesn't make clear what the default locking is if you don't specify LOCK=NONE so it's better to be safe. 

[1] https://dev.mysql.com/doc/refman/8.0/en/innodb-online-ddl-operations.html#online-ddl-primary-key-syntax-notes
Search for "Dropping a primary key and adding another"

Reply:
Use Algorithm=Inplace, LOCK=NONE here and also for drop columns. Tested locally with fake prod db , we can write to Invocations, Targets, TargetStatuses during schema change.